Refresh clinetic synthea fork

epeden opened the pull request.

eedrummer committed:
WIP: Starting update on UTI with telemedicine and ED paths

eedrummer committed:
More UTI update implementation

eedrummer committed:
WIP: more UTI update implementation

eedrummer committed:
WIP: Adding labs to UTI module

eedrummer committed:
WIP: UTI lab follow up added

eedrummer committed:
WIP: Updating UTI lab bacteria to be set via attribute

eedrummer committed:
WIP: Finished UTI ED Bundle submodule

eedrummer committed:
WIP: Implementing UTI antibiotics submodule

eedrummer committed:
WIP: UTI update cleanup and debugging

eedrummer committed:
WIP: Yet more debugging of uti updates

eedrummer committed:
WIP: Fixing bug in Cystitis onset and bug in UTI recurrence not setting the actual condition

eedrummer committed:
Adding UTI as the reason to the encounters and medications

eedrummer committed:
Fixing a typo in distro probability and fixing Ceftriaxone single dose

eedrummer committed:
Java and UTI module fixes

Some parts of the telemedicine support were crashing the test suite as they
haven't been used before.

I also missed an entire slide of module implementation in the UTI telemed submodule.

eedrummer committed:
Addressing an issue on the UTI ED path with a temp fix

eedrummer committed:
Final tweak to the UTI ED Eval submodule to handle lack of information on inpatient admission

jawalonoski committed:
Add testing module to ensure HHA visits for RIF unit tests.

hadleynet committed:
Allow valid HICN in addition to synthetic. Change visibility of HICN property for re-use in X12 exporter.

eedrummer committed:
Adjustment to lookUp to prevent issues when close to 100th percentile.

dehall committed:
fix export filter keeping old entries when there is an active entry of the same type

pramod2804 committed:
Create CloudFormation.yaml

Initial commit

eedrummer committed:
Fix issues with NPEs related to the employment module

The simulation expects every person to have an employment model set in
their attributes. If a person was generated an a county where the
`sdoh.csv` file did not have a row, the employment model would not get
set. This change uses the state average information when the desired
county can not be found.

eedrummer committed:
Fix for NPE seen when sometimes trying to clone the next state

Issue appears to be related to having two submodules with the same name
as a part of a module. This renames the submodules to have unique names.

arvindshmicrosoft committed:
Replace SimpleDateFormat with FastDateFormat in ExportHelper

ExportHelper originally used SimpleDateFormat, which is not thread-safe.
Hence access to the static DATE_FORMAT and ISO_DATE_FORMAT had to be
protected using `synchronized`, as noted in the original code.

This synchronized access is especially visible on higher end machines.
For example on a VM with 48 vCPUs, a `jstack -l` output analysis showed
almost 50% of the pool threads were waiting on the static `ISO_DATE_FORMAT`
instance within ExportHelper. On such machines, CPU usage would be low
and Synthea's throughput was not scaling well with higher number of vCPUs.

This change safely replaces those usage of SimpleDateFormat with
FastDateFormat, which is thread-safe, thereby removing the need to
synchronize access to those static instances across threads. The result of
this is better scalability and improved usage of CPU, especially on
higher-end compute.

dehall committed:
checkstyle

arvindshmicrosoft committed:
Address checkstyle issues

eedrummer committed:
Merge pull request #1388 from arvindshmicrosoft/use-fastdateformat

Replace SimpleDateFormat with FastDateFormat in ExportHelper

dehall committed:
Merge pull request #1386 from synthetichealth/fix-employment-model-npe

Fix issues with NPEs related to the employment module

awspramod committed:
Merge branch 'synthetichealth:master' into master

dehall committed:
fill in an obvious TODO

eedrummer committed:
Update src/main/java/org/mitre/synthea/world/concepts/GrowthChartEntry.java

Co-authored-by: Dylan Hall <dehall@mitre.org>

dehall committed:
Merge pull request #1381 from synthetichealth/fix-vital-signs-nan

Fix issue where vital signs were sometimes being set to `NaN`

dehall committed:
new script and deploy action step to generate 1k sample patient data

dehall committed:
use the right username

jawalonoski committed:
Merge pull request #1385 from synthetichealth/export-filter-active-repeats

Fix export filter keeping old entries when there is an active entry of the same type

jawalonoski committed:
Merge pull request #1394 from synthetichealth/autogenerate_sample_data

Autogenerate 1k sample patient data files

dehall committed:
delete uncompressed sample data

jawalonoski committed:
Merge pull request #1395 from synthetichealth/fix-deploy

Delete uncompressed sample data in deploy workflow

jawalonoski committed:
Merge pull request #1393 from awspramod/master

Synthea deployment on AWS using cloudformation template

dehall committed:
update RandomCodeGenerator to read from HAPI ValueSet instead of JSON model

dehall committed:
reduce samples to 100 patients

jawalonoski committed:
Change final status of NPI for X12 exporter.

jawalonoski committed:
Merge pull request #1387 from synthetichealth/fix-referral-npe

Fix for NPE seen when sometimes trying to clone the next state

eedrummer committed:
Update src/main/resources/modules/uti/abx_tx.json

Co-authored-by: Dylan Hall <dehall@mitre.org>

eedrummer committed:
Update src/main/resources/modules/uti/abx_tx.json

Co-authored-by: Dylan Hall <dehall@mitre.org>

eedrummer committed:
Update src/main/resources/modules/uti/abx_tx.json

Co-authored-by: Dylan Hall <dehall@mitre.org>

eedrummer committed:
Update src/main/resources/modules/uti/ed_path.json

Co-authored-by: Dylan Hall <dehall@mitre.org>

eedrummer committed:
Update src/main/resources/modules/uti/telemed_path.json

Co-authored-by: Dylan Hall <dehall@mitre.org>

eedrummer committed:
Update src/main/resources/modules/uti/telemed_path.json

Co-authored-by: Dylan Hall <dehall@mitre.org>

jawalonoski committed:
Merge pull request #1219 from synthetichealth/flexporter

Flexporter

eedrummer committed:
Removing blank module remarks

jawalonoski committed:
Merge pull request #1261 from synthetichealth/uti-update

UTI Module Update

dehall committed:
ensure appendicitis and rupture conditions are ended

dehall committed:
ensure hypoxemia is ended

dehall committed:
remove 'transition' condition

dehall committed:
no prediabetes diagnosis after diabetes

dehall committed:
ensure overdose conditions are ended

dehall committed:
ensure miscarriage condition is ended, and add history of miscarriage instead

dehall committed:
remove 'transfer' condition

jawalonoski committed:
Fix Oxygen Saturation codes to match US Core requirements.

dehall committed:
bring back reasons for SNF and home health modules

dehall committed:
set reason code only if not already set

dehall committed:
revert home_hospice_snf

dehall committed:
clear out attributes at completion of module

dehall committed:
flexporter enhancements

dehall committed:
templates and keep modules

dehall committed:
rename utilities method

eedrummer committed:
Merge pull request #1403 from synthetichealth/templates

Templates and Keep modules

eedrummer committed:
Merge pull request #1399 from synthetichealth/module_cleanup

Module cleanup

dehall committed:
new script to update code displays

dehall committed:
fix obvious code/display issues

eedrummer committed:
Merge pull request #1402 from synthetichealth/flexporter_qicorenotdone

Flexporter enhancements

dehall committed:
refactor with better snomed handling

hadleynet committed:
Add new dependency needed for X12 exporter

jawalonoski committed:
Merge pull request #1407 from synthetichealth/update_codedisplay_script_js

New script to update code displays

dehall committed:
better handling of lists in flexporter

dehall committed:
testing with primitives

jawalonoski committed:
Read modules from multiple paths, plus dental costs.

jawalonoski committed:
Graphviz and module override fixes.

dehall committed:
add support for code system versions

jawalonoski committed:
Merge pull request #1434 from synthetichealth/code_versions

Add support for code system versions

jawalonoski committed:
Change transition metrics to export to JSON.

jawalonoski committed:
Performance adjustment.

jawalonoski committed:
Merge pull request #1422 from synthetichealth/flexporter_lists

Better handling of lists in Flexporter

jawalonoski committed:
Convert String predicate to ModuleSupplier predicate.

hadleynet committed:
Add dependency needed by X12 exporter

jawalonoski committed:
Fix merge conflicts.

jawalonoski committed:
Reattached migrated method.

jawalonoski committed:
Fix #1436, covid vaccine descriptions.

nickpolliard816 commented:
We are probably good to go here once we get through the Jefferson stuff. But I do think this will change the "Standard Population" that we generated once already. No big deal but we will have to re run it, and then reload it.

nickpolliard816 commented:
We are probably good to go here once we get through the Jefferson stuff. But I do think this will change the "Standard Population" that we generated once already. No big deal but we will have to re run it, and then reload it.

dehall committed:
Add support for US Core 3.1.1 and 6.1.0

anikamk1828 committed:
added feature #1326, middle name in patients.csv

jkiddo committed:
Added feature for generating the parameters file

jkiddo committed:
Reverting back to default

jawalonoski committed:
Merge pull request #1451 from anikamk1828/patch-1

Middle Name added to patients.csv

NinaLua committed:
chore: fix some comments

jawalonoski committed:
Merge pull request #1454 from NinaLua/master

chore: fix some comments

jkiddo committed:
Added config option and fixed checkstyle remarks

jawalonoski committed:
Fix #1414.

jawalonoski committed:
Merge pull request #1450 from synthetichealth/uscore6

Add support for US Core 3.1.1 and 6.1.0

dehall committed:
Fix for US Core Pulse Ox error

jawalonoski committed:
Merge pull request #1456 from synthetichealth/pulse_ox_fix

Quick Fix for US Core Pulse Ox error

jawalonoski committed:
Module syntax corrections: transitions.

jawalonoski committed:
Module syntax corrections: removed superfluous names and empty target_encounters.

jawalonoski committed:
Module syntax corrections: remove fields, correct booleans.

jawalonoski committed:
Module syntax corrections: additional touch ups.

jawalonoski committed:
Merge pull request #1452 from trifork/feat/support-bulk-import-format

Added feature for generating the parameters file

jawalonoski committed:
Fix checkstyle complaints.

hadleynet committed:
Fix a bug where setting exporter.years_of_history via the command line would be ignored

hadleynet committed:
Fix checkstyle warnings

jawalonoski committed:
Fix merge conflict.

jawalonoski committed:
Fix unit test loading modules from jar.

jawalonoski committed:
Additional unit tests.

dehall committed:
Remove SHR extensions from FHIR exporters

jawalonoski committed:
Merge pull request #1472 from synthetichealth/remove_shr

Remove SHR extensions from FHIR exporters

jkiddo committed:
Update synthea.properties

jawalonoski committed:
Merge pull request #1478 from trifork/master

Update synthea.properties

dehall committed:
Fix issue with setting config via CLI

jawalonoski committed:
Merge pull request #1482 from synthetichealth/fix_config_cli

Fix issue with setting config via CLI

jawalonoski committed:
Changes associated with IDR analysis and updates.

jawalonoski committed:
Change comment to match property name.

jkiddo committed:
Moved generation of file to _after_ the other ndjson files are generated as they need to be referenced as well (organization, practitioner, practitionerrole & location)
Corrected default port to 8000 to follow documentation.

jawalonoski committed:
Merge pull request #1491 from trifork/fix/include-all-files

Moved generation of file to _after_ the other ndjson files are genera…

jkiddo committed:
Update RandomCodeGenerator.java

jawalonoski committed:
Merge pull request #1492 from trifork/jkiddo-patch-1

Adjust headers to work as intended when requesting TX servers

jkiddo committed:
Expansion may fail to supply a display value

jawalonoski committed:
Merge pull request #1493 from jkiddo/patch-1

Expansions from TX may fail to supply a display value. This fixes that null pointer

jkiddo committed:
Update gradle-wrapper.properties

jawalonoski committed:
Fix merge conflicts.

hadleynet committed:
Merge pull request #1357 from synthetichealth/dental

Dental Modules

lmd59 committed:
Fix ids for flexporter executeScript

lmd59 committed:
Add null check to pass test

dehall committed:
Merge pull request #1501 from lmd59/flexporter-jsids

Fix ids for flexporter executeScript

hadleynet committed:
3.3.0 release

hadleynet committed:
3.3.1-SNAPSHOT start

jawalonoski committed:
Changes to hypertension onset to better align with statistics.

Commit:
Updated snomed codes, based on the discusssion with dylan

Commit:
Updated the SNOMED codes to strings

hadleynet committed:
Add support for mapping condition codes to ICD-10-CM in the FHIR R4 exporter. Fix some checkstyle warnings.

dehall committed:
Merge pull request #1520 from florim14/snomedCode_updates

Updated snomed codes, based on the exchange discusssion with Dylan

dehall committed:
support for 'if' for conditionally mapping single fields

hadleynet committed:
Extract method for adding ICD10 translation codes, add to other types of resources that have a reason.

dehall committed:
delete resources by fhirpath

dehall committed:
enable functions nested

dehall committed:
fix checkstyle issues

jawalonoski committed:
Merge pull request #1525 from synthetichealth/flexporter_updates_2024_10

Flexporter updates - October 2024

jawalonoski committed:
Resolve merge conflict.

jawalonoski committed:
Merge pull request #1521 from synthetichealth/icd_code_mapping

ICD-10-CM Code Mapping

dehall committed:
Update randomCode fn to take output type as an optional second param

dehall committed:
add custom ValueSets to flexporter mapping file

Commit:
Updated the remaining Snomed codes from the discussion with Dylan and Carlos

jawalonoski committed:
Small updates to add defaults for missing data or configuration settings, mostly applicable to synthea-international configurations.

dehall committed:
Merge pull request #1516 from synthetichealth/hypertension_onset

Changes to hypertension onset to better align with statistics.

dehall committed:
Merge pull request #1527 from synthetichealth/randomcode_updates

Flexporter - Update randomCode fn to take output type as an optional second param

dehall committed:
Merge pull request #1564 from synthetichealth/international_updates

Defaults for some missing configuration settings

dehall committed:
restore previous dental postop care code

dehall committed:
Merge pull request #1559 from florim14/snomed_code-updates_v2

Updated the remaining Snomed codes from the discussion with Dylan and Carlos

lmd59 committed:
Set MedicationRequest status to COMPLETE instead of STOPPED

lmd59 committed:
Change STOPPED to COMPLETED for previous fhir versions

jawalonoski committed:
Merge pull request #1572 from synthetichealth/medrequest_status

Set MedicationRequest status to COMPLETE instead of STOPPED

dehall committed:
Diabetic retinopathy (#1483)

* initial changes for diabetic retinopathy module, not yet SME reviewed

* cleanup

* review improvements

* model enhancements

* wrap logmar unit in braces {}

* enhancements

* add doc and stage observation codes

* Update diabetic_retinopathy_module_guide.md

* Update diabetic_retinopathy_module_guide.md

* add sample records to repo  so they can be loaded in the record viewer

* quick fixes

* make sure the associate_images script is actually in the repo

* dicom fundus photos

* improvements based on review feedback

* additional updates

* add module remarks

* update docs

* latest updates before rebase

* latest version of the run script

* updated dicom script

* add all python scripts just for reproducibility

* script fixes

* complete run through of run_coherent_eyes

* one last change

* cleanup for merge to master

* little more cleanup, delete files and pluck some more from master

* move keep modules to resources

* add reasons to procedures

facuerrobidart committed:
Allow included and excluded files in CSV exporter if there is no overlap (#1573)

* Allow both included and excluded files on CSVExporter if there is no overlap

* Fix linting on tests

awatson1978 committed:
Merge branch 'patch-2' of https://github.com/jkiddo/synthea into patch-2

jawalonoski committed:
Add labs/reports to clinical notes.

KeeyanGhoreshi committed:
Merge pull request #1585 from synthetichealth/gradle-update

Gradle Update

KeeyanGhoreshi committed:
fix javadoc and checkstyle warnings

KeeyanGhoreshi committed:
minor change

KeeyanGhoreshi committed:
fix jdoc errors

KeeyanGhoreshi committed:
update person/provider

KeeyanGhoreshi committed:
Merge pull request #1589 from synthetichealth/notes_update

Add labs/reports to clinical notes.

KeeyanGhoreshi committed:
fix claim/provider/org serialization

dehall committed:
change randUUID to reduce chances of duplicate uuids in large populations

dehall committed:
update test to account for different RNG calls

jawalonoski committed:
Fix claim transaction and payer transition referential integrity. Fixes #1508.

jawalonoski committed:
Merge pull request #1593 from synthetichealth/warning-fixes

Warning fixes

jawalonoski committed:
Merge branch 'master' into serialization-fixes

jawalonoski committed:
Fix checkstyle warnings.

jawalonoski committed:
Merge pull request #1598 from synthetichealth/serialization-fixes

fix claim/provider/org serialization

Jammjammjamm committed:
WIP add initial CSVFileManager and use it for patients

Jammjammjamm committed:
extract CSV constants to a separate class

Jammjammjamm committed:
move getWriter implementation to CSVFileManager

Jammjammjamm committed:
move header writing to CSVFileManager

Jammjammjamm committed:
store resource writers in CSVFileManager

Jammjammjamm committed:
rename resource export methods to indicate they are exporting something

Jammjammjamm committed:
move writer initialization to write method

Jammjammjamm committed:
update synchronization for csv writers

Jammjammjamm committed:
cleanup

Jammjammjamm committed:
move append initialization to CSVFileManager

Jammjammjamm committed:
move outputDirectory initialization to CSVFileManager

Jammjammjamm committed:
move included/excluded files initialization to CSVFileManager

Jammjammjamm committed:
move CSVFileManager initialization code to separate methods

Jammjammjamm committed:
remove TODO

Jammjammjamm committed:
move write method to CSVFileManager

Jammjammjamm committed:
change flushWriters to only flush a single file

Jammjammjamm committed:
add configuration property for max lines per file

Jammjammjamm committed:
refactor filenames for normal csv output

Jammjammjamm committed:
split csv output across multiple files

Jammjammjamm committed:
add support for appending with multiple csv files

Jammjammjamm committed:
fix linting issues

Jammjammjamm committed:
fix intermittent csv export test failure

Jammjammjamm committed:
add unit test for multiple csv files

Jammjammjamm committed:
fix off-by-one error when determining file count when appending to multiple csv files

Jammjammjamm committed:
add unit test for exporting to multiple files with append mode

Jammjammjamm committed:
update shadow plugin

Jammjammjamm committed:
move remaining plugins to plugins block

Jammjammjamm committed:
move sourceCompatibility to java block

Jammjammjamm committed:
move main class into application block

Jammjammjamm committed:
update docs link to point to current version of gradle

Jammjammjamm committed:
update gradle to 8.14.3

Jammjammjamm committed:
fix new deprecation warnings

Jammjammjamm committed:
update docs link to point to current version of gradle

Jammjammjamm committed:
add fileNumberDigits option

Jammjammjamm committed:
add formatting for file number

Jammjammjamm committed:
move common test setup to before block

Jammjammjamm committed:
add unit test for file number formatting

Jammjammjamm committed:
use us core 6 as default in properties file

jawalonoski committed:
Merge pull request #1609 from synthetichealth/refactor-csv-exporter

Refactor CSVExporter

jawalonoski committed:
Merge pull request #1602 from synthetichealth/fix1508

Fixes #1508

jawalonoski committed:
Remove Slack notifications from build and deploy processes.

jawalonoski committed:
Merge pull request #1617 from synthetichealth/no_slack

Remove Slack notifications from build and deploy processes.

jawalonoski committed:
Merge pull request #1613 from synthetichealth/us-core-6-default

Use US Core 6 as default in properties file

andrequina committed:
Add colored donation link at end of patient generation. The message includes color-coded text and a clickable hyperlink to the GitHub sponsors page.

andrequina committed:
Fix newline issue in Generator.java

jawalonoski committed:
Merge pull request #1601 from synthetichealth/dup_uuid_fix

Reduce chances of duplicate uuids in large populations

jawalonoski committed:
Merge pull request #1610 from synthetichealth/fix-gradle-deprecation-warnings

Upgrade gradle and fix deprecation warnings

jawalonoski committed:
Merge pull request #1618 from synthetichealth/limit-csv-lines

Limit resources per CSV file

andrequina committed:
updated donation message to show a visible link in case the link in the terminal is not clickable.

jawalonoski committed:
Merge pull request #1619 from andrequina/add_donation_link

Add donation message at end of patient generation.

hadleynet committed:
Fix checkstyle warnings

hadleynet committed:
3.4.0 release

hadleynet committed:
3.4.1-SNAPSHOT start

hadleynet committed:
Update copyright dates

Ritakoubeh committed:
Add AIS_From_School_Screening_to_SOSORT_Recommendations.json

Synthetic Synthea module simulating the workflow of Adolescent Idiopathic Scoliosis (AIS) from school screening to SOSORT 2016 treatment recommendations. 
Includes patient encounters, referrals, radiographic assessments, and treatment decisions. Fully synthetic data; customizable for research or simulation purposes.

Contributors: 
- @Ritakoubeh (Reem Koubeh)
- Mohammad Firas Wahbeh
- Daoud Karakola

Ritakoubeh committed:
Update AIS_From_School_Screening_to_SOSORT_Recommendations.json

Add AIS module JSON code

Ritakoubeh committed:
Update AIS_From_School_Screening_to_SOSORT_Recommendations.json

deleted units

Ritakoubeh committed:
Merge pull request #1 from Ritakoubeh/Ritakoubeh-patch-1

Update AIS_From_School_Screening_to_SOSORT_Recommendations.json

Ritakoubeh committed:
Update AIS_From_School_Screening_to_SOSORT_Recommendations.json

Fix error in AIS module JSON

Ritakoubeh committed:
Merge pull request #2 from Ritakoubeh/Ritakoubeh-patch-2

Update AIS_From_School_Screening_to_SOSORT_Recommendations.json

Jammjammjamm committed:
US Core 7 Support (#1620)

* add us core 7 mapping code

* update mappings for us core 7

* add us core 7 conformance resources for testing

* add unit test for us core 7 export

* WIP

* update loinc 787-2 display

* update loinc 786-4 display

* replaced deprecated loinc code

* update loinc 32623-1 display

* update sdoh survey displays

* update cvx 140 display

* update loinc 2571-8 display

* update loinc 18262-6 display

* update display for RxNorm 748879

* update display for RxNorm 1605257

* update display for RxNorm 751905

* update displays in medication costs csv

* update display for loinc 20570-8

* update display for RxNorm 978950

* update display for RxNorm 807283

* revert changes to medications cost csv

* update display for snomed 169553002

* update display for RxNorm 748962

* update remaining contraceptive displays

* update loinc 92142-9 display

* update loinc 92141-1 display

* update loinc 92131-2 display

* update loinc parainfluenza displays

* update display for loinc 92130-4 and 92134-6

* update loinc 94040-3 display

* update loinc 92143-7 display

* update display for RxNorm 831533

* update display for loinc 91148-7

* update loinc 33914-3 display

* fix RxNorm 1367439 display whitespace

* update loinc 788-0 display

* fix RxNorm 1367439 display

* fix RxNorm 198031 display

* fix snomed 90560007 display

* update snomed 415070008 display

* fix RxNorm 284215 display

* fix snomed 5602001 display

* update RxNorm 200346 display

* update snomed 1551000119108 display

* fix snomed 57676002 display

* fix loinc 58410-2 display

* fix RxNorm 108515 display

* fix RxNorm 1664463 display

* update loinc 24321-2 display

* update loinc 51990-0 display

* fix loinc 600-7 display

* update RxNorm 897685 display

* fix RxNorm 1656356 display

* fix RxNorm 2563431 display

* fix RxNorm 205532 display

* fix RxNorm 999999 display

* fix RxNorm 1431987 display

* fix RxNorm 542347 display

* fix RxNorm 1856546 display

* fix RxNorm 748856 display

* fix RxNorm 749762 display

* fix RxNorm 749785 display

* fix RxNorm 1359133 display

* fix RxNorm 1243052 display

* fix loinc 5821-4 display

* fix loinc 13945-1 display

* update loinc 5787-7 display

* fix loinc 24124-0 display

* fix loinc 8247-9 display

* fix loinc 5769-5 display

* fix loinc 75636-1 display

* update loinc 85354-9 display

* fix loinc 788-0 display

* fix loinc 33762-6 display

* fix loinc 33914-3 display

* update loinc 2106-3 display

* fix loinc 85318-4 display

* fix loinc 2339-0 display

* fix loinc 6299-2 display

* fix loinc 38483-4 display

* fix loinc 49765-1 display

* fix loinc 2947-0 display

* fix loinc 6298-4 display

* fix loinc 2069-3 display

* fix loinc 20565-8 display

* update loinc 5767-9 display

* update loinc 770-8 display

* update loinc 736-9 display

* update loinc 5905-5 display

* update loinc 713-8 display

* update loinc 706-2 display

* fix loinc 718-7 display

* fix loinc 30428-7 display

* fix loinc 30385-9 display

* fix loinc 26515-7 display

* fix loinc 42719-5 display

* fix loinc 1742-6 display

* fix loinc 1920-8 display

* fix loinc 33037-3 display

* fix loinc 2885-2 display

* fix loinc 1751-7 display

* fix loinc 10834-0 display

* fix loinc 6768-6 display

* fix bcp47 ru-RU display

* fix RxNorm 808917 display

* fix RxNorm 1648756 display

* fix RxNorm 1997015 display

* fix RxNorm 1648755 display

* fix RxNorm 808917 display

* fix RxNorm 1648755 display

* fix RxNorm 856980 display

* fix RxNorm 212446 display

* fix RxNorm 1946840 display

* fix RxNorm 1648755 display

* fix RxNorm 226719 display

* fix snomed 1501000119109 display

* fix RxNorm 1747691 display

* fix RxNorm 2119714 display

* fix RxNorm 1648756 display

* fix RxNorm 1648755 display

* fix RxNorm 1789958 display

* fix RxNorm 1190766 display

* add pregnancy status observation to pregnancy module

* add pregnancy status to export for us core 6 & 7

* fix assigning pregnancy status profile

* add value_code to pregnancy module

* remove duplicate us core mapping entries for loinc 76690-7

* update category for sexual orientation observation

* update homelessness module to create sexual orientation observations which fit the US Core profile

* update snomed suicide attempt code displays

* update display for RxNorm 999999

* add unknown system to excluded systems to update displays

* update remaining outdated displays

* update RxNorm 999999 to 999970

* update RxNorm 205532 display

dehall committed:
Upgrade to Gradle 9 for JDK 25 support, remove JDK 11 support

dehall committed:
update README

dehall committed:
Run gradle wrapper task to get latest version of wrapper files

dehall committed:
Fix one new deprecation warning

Ritakoubeh committed:
Merge branch 'synthetichealth:master' into master

daoudkarakola1974 committed:
Recommendations Implemented

Ritakoubeh committed:
Update AIS_From_School_Screening_to_SOSORT_Recommendations.json

AIS-Module-Enhanced

dehall committed:
Merge pull request #1631 from Ritakoubeh/master

Create AIS_From_School_Screening_to_SOSORT_Recommendations.json

dehall committed:
use withSourcesJar and withJavadocJar instead of archives and manually defined tasks

hadleynet committed:
4.0.0 release

hadleynet committed:
4.0.1-SNAPSHOT start

hadleynet committed:
Reorder condition onset and diagnosing encounter in asthma module

Commit:
Cache parsed range bounds in Demographics to avoid repeated regex/parseInt

The pickAge() and pickIncome() methods are called for every generated
patient. Previously, each call performed a regex split on the range
string (e.g., '0..4') and Integer.parseInt() on both parts. Since the
set of range strings is small and fixed (defined by the demographics
CSV), this parsing work is entirely redundant after the first call.

This change introduces a static ConcurrentHashMap cache that stores
the parsed int[low, high] bounds for each range string. The cache is
populated lazily via computeIfAbsent and is thread-safe for parallel
patient generation.

For a typical run generating 10,000 patients, this eliminates ~20,000
redundant regex operations and ~40,000 Integer.parseInt calls.

Addresses the TODO comments in pickAge() and pickIncome():
  'this seems like it would benefit from better caching'

Commit:
Implement sex-specific NHANES sampling for pediatric growth trajectories

Previously, the PediatricGrowthTrajectory selected initial BMI samples
from the full NHANES dataset without considering the child's sex. Since
male and female children exhibit different BMI distributions and growth
patterns (CDC growth charts are sex-specific), this produced less
clinically realistic synthetic patient data.

This change:
- Adds NHANESSample.loadDistributionBySex(int sexCode) which filters
  NHANES samples by sex and re-normalizes probabilities within the
  filtered set
- Adds a new PediatricGrowthTrajectory(seed, birthTime, sex) constructor
  that selects from the sex-appropriate distribution
- Updates LifecycleModule to pass the person's gender to the constructor
- Maintains backward compatibility via the deprecated original constructor

The NHANES dataset uses sex codes 1 (male) and 2 (female). The filtered
distributions are loaded once at class initialization and are thread-safe
for parallel patient generation.

This improves the clinical realism of generated pediatric BMI data,
particularly for studies examining sex-specific obesity prevalence and
growth patterns in synthetic populations.

Addresses the TODO in PediatricGrowthTrajectory:
  'Make the selection sex specific'

bobu-putheeckal committed:
Handle long help argument without NPE

bobu-putheeckal committed:
Avoid max timestamp in CCDA coverage fallback

bobu-putheeckal committed:
Use exact decimal for employer premium share

bobu-putheeckal committed:
Use interval for CCDA vital signs organizer

jawalonoski committed:
Merge pull request #1660 from bobu-putheeckal/fix-long-help-arg

Handle --help without NullPointerException

jawalonoski committed:
Merge pull request #1658 from RohithVangalla1/fix/cache-demographics-range-parsing

Cache parsed range bounds in Demographics to avoid repeated regex/parseInt per patient

jawalonoski committed:
Merge pull request #1663 from bobu-putheeckal/fix-1425-premium-decimal-format

Use exact decimal for employer premium share

jawalonoski committed:
Merge pull request #1662 from bobu-putheeckal/fix-1360-ccda-coverage-stop-time

Avoid max timestamp in CCDA coverage fallback

jawalonoski committed:
Merge pull request #1664 from bobu-putheeckal/fix-995-ccda-vital-signs-interval

Use interval for CCDA vital signs organizer

Commit:
Add tests for sex-specific NHANES distributions and PediatricGrowthTrajectory

Added tests for NHANESSample.loadDistributionBySex():
- loadDistributionBySexMale: verifies all samples from male distribution
  have sex code 1
- loadDistributionBySexFemale: verifies all samples from female distribution
  have sex code 2
- loadDistributionBySexProbabilitiesNormalized: verifies both distributions
  produce valid samples with positive BMI (confirms re-normalization works)
- sexSpecificDistributionsAreDeterministic: verifies same seed produces
  same sample for reproducibility

Added tests for PediatricGrowthTrajectory(seed, birthTime, sex):
- sexSpecificConstructorMale: verifies male trajectory has valid BMI and
  age in expected range (24-36 months)
- sexSpecificConstructorFemale: same verification for female
- sexSpecificConstructorIsDeterministic: same seed+sex produces same BMI
- sexSpecificConstructorDiffersBySex: male and female trajectories are
  both valid (draw from different distributions)
- sexSpecificConstructorGeneratesTrajectory: verifies the sex-specific
  trajectory can generate future BMI values via currentBMI()

jawalonoski committed:
Merge pull request #1659 from RohithVangalla1/fix/sex-specific-pediatric-growth-nhanes-sampling

Implement sex-specific NHANES sampling for pediatric growth trajectories

javabrett committed:
Fix timezone-dependent failure in receiveDualEligible test

Replace Calendar.getInstance() (local timezone) with
Utilities.convertCalendarYearsToTime() (UTC) for the age-64 and
age-65 boundary timestamps in PayerTest.receiveDualEligible.

The local Calendar.add() call would land on Dec 31 in UTC on hosts
whose timezone has a different UTC offset between 1950 and 2015 due
to DST history (e.g. Australia/Sydney). Person.age() computes
entirely in ZoneOffset.UTC, so the resulting instant was seen as age
64 rather than 65, causing MedicareEligible to fail and the person
to receive Medicaid instead of Dual Eligible.

The fix uses the same UTC basis as birthTime and the ages-0-64 loop
already do, making the test timezone-independent.

Fixes #1530

rtmalikian committed:
docs: document argument order precedence for -c and --config flags

When both -c (config file) and --config*=value (command-line override)
are used, the last one specified wins because they write to the same
Properties object. This adds a note to the usage help text explaining
this behavior with concrete examples.

Fixes #1668

Signed-off-by: Raphael Malikian <rtmalikian@gmail.com>

rtmalikian committed:
fix: set Claim.total to sum of item.net values in FHIR exports

The FHIR Claim.total was set to the post-adjustment claim cost
(getTotalClaimCost()), but item.net values used pre-adjustment raw
costs from entry.getCost(). This caused Claim.total to not equal
the sum of item.net values, violating the FHIR specification.

Additionally, the encounter item (item 1) was missing a net value
entirely, even though the encounter cost was included in the total.

Changes:
- Add encounter cost as net value on item 1 in both FhirR4 and FhirStu3
- Track sum of all item.net values as items are added
- Use the accumulated sum for Claim.total instead of getTotalClaimCost()

Fixes #1513

Signed-off-by: Raphael Malikian <rtmalikian@gmail.com>

rtmalikian committed:
fix: use .equals() instead of == for String comparisons

Replace reference equality (==) with value equality (.equals()) for
String comparisons in three files:

- BirthStatistics.java: use "M".equals() for gender check to avoid
  silent failure when attributes contain non-interned strings
- CPCDSExporter.java: use .equals() for networkStatus and
  sourceAdminCode comparisons in place-of-service logic
- State.java: use .isEmpty() instead of == "" for altDirectTransition

Fixes #1677

Signed-off-by: Raphael Malikian <rtmalikian@gmail.com>

rtmalikian committed:
fix: correct PDW lab value range from 150-520 to 9-17 fL (Fixes #1557)

The Platelet Distribution Width (LOINC 32207-3) range was incorrectly set
to 150-520 fL, clearly copied from the adjacent Platelet Count (LOINC 777-3,
range 150-450 in 10*3/uL). The correct PDW range is 9-17 fL based on
published reference values (PMC2843567).

Fixed in 9 module JSON files where PDW observations are defined.

Signed-off-by: Raphael Malikian <rtmalikian@gmail.com>

rtmalikian committed:
fix: use null-safe String comparison for observation.category in FhirR4

observation.category.equals() was called at lines 1876, 1879, and 2555
without null guards, while line 1884 already checks for null before
using observation.category in a switch statement. When observation.category
is null, these calls throw NullPointerException.

Fix by using the idiomatic Java null-safe pattern: put the String literal
on the left side of .equals()/.equalsIgnoreCase(). This returns false
when the argument is null instead of throwing NPE.

Fixes #1680

Signed-off-by: Raphael Malikian <rtmalikian@gmail.com>

rtmalikian committed:
fix: close FileInputStream in Mapping.parseMapping() via try-with-resources

The FileInputStream created in parseMapping() was never closed,
causing a resource leak. Wrapped the stream in a try-with-resources
block to ensure it is always closed after yaml.loadAs() completes.

Also updated ActionsTest.setupClass() throws clause from
FileNotFoundException to IOException to match the widened signature.

Fixes #1682

Signed-off-by: Raphael Malikian <rtmalikian@gmail.com>

rtmalikian committed:
fix: guard against empty codes list in determineCostOfEntry

The testFHIRExportExcludes test triggers a claim export where the
encounter entry has an empty codes list. The new Claim.total
calculation calls getCost() which triggers determineCostOfEntry(),
which crashes with IndexOutOfBoundsException on codes.get(0) when
the list is empty.

Fix: return 0.0 cost when codes list is empty.
Signed-off-by: rtmalikian <rtmalikian@gmail.com>

rtmalikian committed:
fix: return default cost (not zero) when entry has no codes

Per reviewer feedback: when entry.codes is empty, return the
default cost * locationAdjustment instead of 0.0. This matches
the behavior for items that don't have code-specific costs.

Signed-off-by: rtmalikian <rtmalikian@gmail.com>

dehall committed:
Merge pull request #1676 from rtmalikian/fix/issue-1513-claim-total-mismatch

fix: set Claim.total to sum of item.net values in FHIR exports (Fixes #1513)

reachpratyuk-dotcom committed:
fix Patient identifiers (SSN, DRIVERS, PASSPORT) are not unique — collisions at population scale

reachpratyuk-dotcom committed:
fix Patient identifiers ( PASSPORT) are not unique — collisions at population scale

reachpratyuk-dotcom committed:
fix Patient identifiers (SSN, DRIVERS, PASSPORT) are not unique — collisions at population scale

reachpratyuk-dotcom committed:
fix Patient identifiers (PASSPORT) are not unique — collisions at population scale

reachpratyuk-dotcom committed:
fix checkstyle warnings

reachpratyuk-dotcom committed:
fix checkstyle warnings

tyre committed:
Honor the configured code system when adding FHIR code translations

FhirR4.addTranslation took a code system, used it to look up a CodeMapper,
and then always emitted the ICD-10-CM system URI. It also only ever asked
for the ICD-10-CM mapper, so a map registered under any other
exporter.code_map.<system> key was loaded at startup and never consulted,
even though Exporter#loadCodeMappers documents that usage and gives CPT as
its example.

Apply every configured code map and label each translation with its own
system URI. Output for the shipped ICD-10-CM configuration is unchanged.

Also add the CPT system URI, match system names case-insensitively so that
names derived from configuration keys resolve, and hold the code mappers in
a TreeMap so that multiple maps are applied in a deterministic order.

tyre committed:
Add NPI columns to organizations.csv and providers.csv

Synthea assigns every organization and clinician an NPI with a valid CMS
check digit, but the CSV exporter writes only internal UUIDs. Clinician
NPIs reach the FHIR Practitioner resource, organization NPIs reach no
exporter at all, and neither reaches CSV, so a consumer working from the
CSV export cannot identify a provider by NPI.

Append an NPI column to both files. The column goes at the end of each row
so that readers that index by position are unaffected.

Also correct the stale header comments on exportOrganization and
exportProvider, which had not been updated when LAT and LON were added.

tyre committed:
Correct the cost of psychotherapy and psychological assessment

SNOMED 228557008 (Cognitive and behavioral therapy) and 405783006
(Psychological assessment) were both priced at 5132.80/10265.59/20531.19.
Both modules that use 228557008 emit it as a single Procedure of 30 to 60
minutes, so a therapy session was costing between five and twenty thousand
dollars. The neighbouring row for psychodynamic interpersonal therapy,
304822001, is 69.20/199.75/330.29.

The values date to 9408e06 (2018), "update procedure cost min/max with 50%
and 200% of mean, just to get something in there", which fanned out an
existing point estimate of 10265.59. That estimate is consistent with a
mean inpatient stay charge rather than an outpatient session price.

Reprice both rows against published fee schedules and cite the sources in
the COMMENTS column, following the convention of the rest of the file:

  228557008  78.93 / 139.00 / 335.00   one 30-60 minute session
  405783006 166.91 / 450.00 / 1200.00  one evaluation

405783006 appears only as a CarePlan activity today and so emits no cost,
but it carried the same placeholder and is corrected alongside.
